add `Job` token type and allocation id

elv-Moritz opened the pull request:
- Introduce a new `Job` token type with allocation and node ID
- Enhance tests to include coverage for `Job` tokens and allocation id

elv-Moritz committed:
add `Job` token type and allocation id

- Introduce a new `Job` token type with allocation and node ID
- Enhance tests to include coverage for `Job` tokens and allocation id

elv-Moritz committed:
update example function names to follow Go naming conventions 1.24

lukseven commented on format/id/id_test.go:
```diff
@@ -15,23 +15,27 @@ var tid = ID(append([]byte{1}, []byte{0, 1, 2, 3, 4, 5, 6}...))
-	generated := Generate(User)
-	assert.NoError(t, generated.AssertCode(User))
+	for code, prefix := range codeToPrefix {
 
```
Remove the empty line.

lukseven commented on format/id/id_test.go:
Please add a test verifying that all internal maps (`prefixToCode`, `codeToPrefix` and `codeToName`) are consistent, i.e. they have the same number of entries, and have entries for each code.

lukseven commented on format/token/token.go:
```diff
@@ -160,6 +180,7 @@ const (
 	QPartWrite        // 2nd version: scheme, flags, random bytes
 	LRO               // node ID, random bytes
 	LocalFile         // node ID, content ID, bytes = hash(offering + presentation + format + start + end)
+	Job
```
```suggestion
	Job               // node ID, allocation ID, bytes = index
```

lukseven reviewed (commented):
Looks good. A couple of minor cleanups and a missing test.

elv-Moritz committed:
add internal maps test and comment for NewJob function

- add `TestInternalMaps` to verify consistency and integrity of prefix and code mappings.
- add a comment to `NewJob` function explaining job token creation.

elv-serban commented on format/drm/drm_test.go:
```diff
@@ -110,7 +110,7 @@ func TestString(t *testing.T) {
 }
 
-func ExampleString() {
+func Example_string() {
```
Is this a new naming convention?

elv-Moritz committed:
update example function name

elv-Moritz committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into mo/allocation

# Conflicts:
#	format/drm/drm_test.go
#	util/ioutil/repeating_reader_test.go
#	util/traceutil/traceutil_multi_goroutine_test.go

lukseven committed:
fix Token.Equal: add allocation ID check

elv-Moritz committed:
Merge remote-tracking branch 'origin/master' into mo/allocation

lukseven committed:
add utilities for partition matching and indexing + tests

lukseven committed:
address code review

elv-Moritz committed:
Merge remote-tracking branch 'origin/luk/partition-util' into mo/allocation

lukseven committed:
address code review

lukseven committed:
Merge remote-tracking branch 'origin/master' into mo/allocation
